core/templates/domain/collection/collection-playthrough.model.ts

```diff
@@ -19,17 +19,18 @@
 import cloneDeep from 'lodash/cloneDeep';
 
 export interface CollectionPlaythroughBackendDict {
-  'next_exploration_id': string;
+  'next_exploration_id': string | null;
```

Comment:
Why `null` is added? Is there a way to refactor the code so that it is not needed?

Reply:
Ditto in all other places.

Reply:
For `CollectionPlaythrough` in spec here

```typescript
it('should return collection is not finished when a next' +
    'exploration is not available', () => {
    const collectionPlaythroughBackendDict = {
      next_exploration_id: null,
      completed_exploration_ids: ['exp_id1']
    };
    const collectionPlaythroughObject = CollectionPlaythrough
      .createFromBackendObject(collectionPlaythroughBackendDict);
    expect(collectionPlaythroughObject.hasFinishedCollection())
      .toEqual(true);
  });
```
we can not set `next_exploration_id` to empty string coz `hasFinishedCollection()` checks it with `null` and test fails. So to pass `null` we have to make the type `string|null` and when we do that we have to do it everywhere in that file.

Reply:
Hmm, if it is just needed for the tests we can add an `@ts-expect-error` in those places. Basically, test code should be always modified to conform to the type instead of the opposite.

As I said, please check all the other places where you set `null` too.

Reply:
For  `CollectionRights` here
```typescript
static createEmptyCollectionRights(): CollectionRights {
    return new CollectionRights({
      owner_names: [],
      collection_id: null,
      can_edit: null,
      can_unpublish: null,
      is_private: null
    });
  }
```
we are assigning `null` to all variable which we can't do coz they are of type `number` and `boolean` so for `number` we have to make it `number|null` 
And for `boolean` if we do `!!null` but it assigns `false` and then tests fail coz we check `toBeNull()` here
```typescript
it('should create an empty collection rights object', () => {
    var emptyCollectionRightsBackendObject = (
      CollectionRights.createEmptyCollectionRights());

    expect(
      emptyCollectionRightsBackendObject.getCollectionId()).toBeNull();
    expect(emptyCollectionRightsBackendObject.canEdit()).toBeNull();
    expect(emptyCollectionRightsBackendObject.canUnpublish()).toBeNull();
    expect(emptyCollectionRightsBackendObject.isPrivate()).toBeNull();
    expect(emptyCollectionRightsBackendObject.getOwnerNames()).toEqual([]);
  });
```
And again if we add `|null` in the definition we have to do it everywhere in that file or it gives errors.

Reply:
Thanks, great analysis, lets add `null` to the `CollectionRights` then.

Reply:
Took a quick look and actually even in this model on line 85 we depend on the `null` type, so it should be fine to add it here too.